src/arch/x86_64/kernel/mod.rs

```diff
@@ -481,7 +481,7 @@ unsafe fn pre_init(boot_info: &'static mut BootInfo) -> ! {
 
 	BOOT_INFO = boot_info as *mut BootInfo;
 
-	if boot_info.cpu_online == 0 {
+	if intrinsics::atomic_load_acq(&(*BOOT_INFO).cpu_online) == 0 {
```

Comment:
Maybe you should use https://github.com/hermitcore/libhermit-rs/blob/master/src/arch/x86_64/kernel/mod.rs#L254

Reply:
I tried atomically loading with acquire on purpose, but this does not seem the problem. :thinking: